presto-main/src/main/java/io/prestosql/operator/project/InputPageProjection.java

```diff
@@ -59,13 +60,28 @@ public InputChannels getInputChannels()
         Block block = requireNonNull(page, "page is null").getBlock(0);
         requireNonNull(selectedPositions, "selectedPositions is null");
 
-        Block result;
-        if (selectedPositions.isList()) {
-            result = block.copyPositions(selectedPositions.getPositions(), selectedPositions.getOffset(), selectedPositions.size());
+        if (isNotLoadedLazyBlock(block)) {
```

Comment:
Also there is more pragmatic reason. If computations were always delayed until needed in `InputPageProjection` then small lazy pages wouldn't be merged in `MergePages` (https://github.com/prestosql/presto/pull/1075/files#diff-bfb4a244d0bf00272b9c34d422388e3aR177) which would affect performance when `work_processor_pipelines` are disabled.
When `work_processor_pipelines` are enabled it also seems to be better to filter loaded block than avoid merging of pages because of https://github.com/prestosql/presto/pull/1075/files#diff-bfb4a244d0bf00272b9c34d422388e3aR177.
When https://github.com/prestosql/presto/pull/1075/files#diff-bfb4a244d0bf00272b9c34d422388e3aR177 is implemented I think we could return lazy identity projections for here for loaded source blocks too.